src/main/kotlin/org/rust/cargo/project/CargoProjectDescription.kt

```diff
@@ -142,6 +142,13 @@ class CargoProjectDescription private constructor(
 
     fun findPackage(name: String): Package? = packages.find { it.name == name }
 
+    fun findLibTargetByNormName(normName: String): Pair<Package, Target>? =
+        packages.asSequence()
+            .mapNotNull { pkg ->
+                pkg.libTarget?.let { if (it.normName == normName) Pair(pkg, it) else null }
+            }
+            .firstOrNull()
```

Comment:
It happened that one of the projects I use to check this feature is [ripgrep](https://github.com/BurntSushi/ripgrep). Besides the executable, it includes five nested libraries that are used as local dependencies (referenced by relative paths, so have `source = null`), but all of them are published on crates.io and have documentation on docs.rs.

Though, we can ignore such cases since we're the authors of the crates and so don't need external documentation for them.